react/components/MiniCartContent.js

```diff
@@ -76,22 +80,27 @@ class MiniCartContent extends Component {
       0
     )
 
-  handleItemRemoval = async ({ id, cartIndex }, index) => {
+  handleItemRemoval = async (item, index) => {
     const { updateItems, updateLocalItems } = this.props
     const updatedItems = [
       {
-        id,
-        index: cartIndex != null ? cartIndex : index,
+        id: item.id,
+        index: item.cartIndex != null ? item.cartIndex : index,
         quantity: 0,
       },
     ]
 
     try {
-      if (cartIndex != null) {
+      if (item.cartIndex != null) {
         await updateItems(updatedItems)
       } else {
         await updateLocalItems(updatedItems)
       }
+
+      this.props.push({
+        event: 'removeFromCart',
+        items: [mapCartItemToPixel(item)],
+      })
```

Comment:
:thinking: is it likely for the `push` function to fail? (or the `mapCartItemToPixel`, for that matter)

Reply:
(asking because it's in a try block that triggers a toast if it fails. so if the "addToCart" function works but the push fails, it would show an error to the user that's maybe unnecessary)

Reply:
I don't think so. The catch doesnt trigger the toast though, just a console.error